src/lib.rs

```diff
@@ -47,13 +50,15 @@ pub fn desugar_book(book: &mut Book, opt_level: OptimizationLevel) -> Result<Def
   book.check_unbound_vars()?;
   book.make_var_names_unique();
   book.linearize_vars();
-  book.eta_reduction(opt_level >= OptimizationLevel::Heavy);
-  book.detach_supercombinators(main);
-  if opt_level >= OptimizationLevel::Heavy {
+  book.eta_reduction(eta);
+  if supercombinators {
+    book.detach_supercombinators(main);
+  }
+  if ref_to_ref {
     book.simplify_ref_to_ref()?;
   }
   book.simplify_main_ref(main);
```

Comment:
This could also be an option. There isn't any reason to disable it really, but I also don't see why not give the option

Reply:
Should it be under the ref-to-ref option or a new one?

Reply:
Isn't it necessary for correct output? For example it fixed this [task](https://app.shortcut.com/hocomp/story/350/fix-wrong-invalid-readback-from-string).
Or are we okay with having some wrong/inconsistent results with the user wants to disable a flag? @developedby

Reply:
Results will still be correct, but readback will be strange.
Also, we shouldn't give an error when a string is not properly terminated since that could be on purpose. Instead it should just readback the term without converting into a string term

Reply:
Also, we are already ok with results being wrong by disabling flags since we have one for ref_to_ref and one for supercombinators